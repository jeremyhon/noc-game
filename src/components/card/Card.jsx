import React from 'react'
import PropTypes from 'prop-types'
import { Card as MuiCard, CardTitle, CardText } from 'material-ui/Card'
import { SkillsContainer, skillsPropType } from '../stats'
import './card.css'

const Card = (props) => (
  <MuiCard className={`card ${props.className}`}>
    <CardTitle style={{padding: 0}} title={props.title} />
    <CardText style={{padding: 0}}>
      <SkillsContainer subtitle={props.subtitle} skills={props.skills} />
    </CardText>
  </MuiCard>
)

Card.defaultProps = {
  className: ""
}

Card.propTypes = {
  className: PropTypes.string,
  title: PropTypes.node.isRequired,
  subtitle: PropTypes.string.isRequired,
  skills: skillsPropType.isRequired,
}

export default Card
