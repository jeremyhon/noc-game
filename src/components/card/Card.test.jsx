import React from 'react'
import { shallow } from 'enzyme'

import Card from './Card'

it("matches the snapshot", () => {
  const component = (
    <Card title="hello"
      subtitle="world"
      skills={{ business: 1, design: 2, technical: 3 }}
    />
  )
  expect(shallow(component)).toMatchSnapshot()
})
