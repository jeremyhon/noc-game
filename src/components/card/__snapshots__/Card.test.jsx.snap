// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<Card
  className="card "
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardTitle
    style={
      Object {
        "padding": 0,
      }
    }
  />
  <CardText
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <SkillsContainer
      className=""
    />
  </CardText>
</Card>
`;
